dockstore-client/src/main/java/io/dockstore/common/FileProvisioning.java

```diff
@@ -270,6 +274,16 @@ private void provisionInputFile(String parameterFilePath, String targetPath, Pat
         URI objectIdentifier = URI.create(targetPath);    // throws IllegalArgumentException if it isn't a valid URI
         if (objectIdentifier.getScheme() != null) {
             String scheme = objectIdentifier.getScheme().toLowerCase();
+            for (PreProvisionInterface plugin : preProvisionPlugins) {
+                List<String> list = plugin.prepareDownload(targetPath);
+                Optional<ImmutablePair<String, String>> newTarget = findSupportedTargetPath(plugins, list);
```

Comment:
Optionally, even though I started the code this way when I first passed it off to you (because I was too lazy to write it), you could have a new class that contains the target and scheme, which is a more readable than a pair of strings.